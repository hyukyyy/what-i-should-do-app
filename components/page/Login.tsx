import {Text} from 'react-native-paper';
import React, {useState} from 'react';
import {StyleSheet, TouchableOpacity, View} from 'react-native';
import Background from '../atom/TextInput.tsx';
import TextInput from '../atom/TextInput.tsx';
import Header from '../atom/Header.tsx';
import {theme} from '../style/themes/theme.ts';
import Button from '../atom/Button.tsx';
import {emailValidator} from '../../utils/validator/emailValidator.ts';
import {passwordValidator} from '../../utils/validator/passwordValidator.ts';
import Logo from '../atom/Logo.tsx';
import {httpPaths, httpPost} from '../../api/httpClient.ts';
import {useDispatch} from 'react-redux';
import {setAccessToken} from '../../redux/reducer/User.ts';
import {useNavigation} from '@react-navigation/native';

const Login = () => {
  const navigation = useNavigation();
  const dispatch = useDispatch();

  const [email, setEmail] = useState({value: '', error: ''});
  const [password, setPassword] = useState({value: '', error: ''});

  const onLoginPressed = async () => {
    const emailError = emailValidator(email.value);
    const passwordError = passwordValidator(password.value);
    if (emailError || passwordError) {
      setEmail({...email, error: emailError});
      setPassword({...password, error: passwordError});
      return;
    }

    try {
      const loginRes = await httpPost(
        httpPaths.login,
        {},
        {
          username: email.value,
          password: password.value,
        },
        '',
      );
      console.log(loginRes);

      if (loginRes.data.token !== null && undefined !== loginRes.data.token) {
        dispatch(setAccessToken({accessToken: loginRes.data.token}));
        navigation.reset({
          index: 0,
          routes: [{name: 'Main' as never}],
        });
      }
    } catch (e) {
      console.error('Error while login : ', e);
    }
  };

  return (
    <Background>
      <>
        <Logo />
        <Header>Welcome back.</Header>
        <TextInput
          label="Email"
          returnKeyType="next"
          value={email.value}
          onChangeText={(text: any) => setEmail({value: text, error: ''})}
          error={!!email.error}
          errorText={email.error}
          autoCapitalize="none"
          autoCompleteType="email"
          textContentType="emailAddress"
          keyboardType="email-address"
          testID={'usernameInput'}
        />
        <TextInput
          label="Password"
          returnKeyType="done"
          value={password.value}
          onChangeText={(text: any) => setPassword({value: text, error: ''})}
          error={!!password.error}
          errorText={password.error}
          secureTextEntry
          testID={'passwordInput'}
        />
        <View style={styles.forgotPassword}>
          <TouchableOpacity
            onPress={() => navigation.navigate('ResetPassword' as never)}>
            <Text style={styles.forgot}>Forgot your password?</Text>
          </TouchableOpacity>
        </View>
        <Button
          mode="contained"
          testID={'loginButton'}
          onPress={onLoginPressed}>
          Login
        </Button>
        <View style={styles.row}>
          <Text>Don't have an account? </Text>
          <TouchableOpacity
            onPress={() => navigation.navigate('Signup' as never)}>
            <Text style={styles.link}>Sign up</Text>
          </TouchableOpacity>
        </View>
      </>
    </Background>
  );
};

const styles = StyleSheet.create({
  forgotPassword: {
    width: '100%',
    alignItems: 'flex-end',
    marginBottom: 24,
  },
  row: {
    flexDirection: 'row',
    marginTop: 4,
  },
  forgot: {
    fontSize: 13,
    color: theme.colors.secondaryText,
  },
  link: {
    fontWeight: 'bold',
    color: theme.colors.primary,
  },
});

export default Login;
