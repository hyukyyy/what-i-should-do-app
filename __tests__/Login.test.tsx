/**
 * @format
 */

import 'react-native';

import '@testing-library/jest-native/extend-expect';
import {expect} from '@jest/globals';
import {renderWithProviders} from '../utils/test/test-utils.tsx';
import {fireEvent} from '@testing-library/react-native';
import Login from '../components/page/Login.tsx';

describe('Login Screen', () => {
  let screen: any;

  // 테스트 작성시 마다 새로 컴포넌트 렌더링
  beforeEach(() => {
    screen = renderWithProviders(<Login />);
  });

  it('renders correctly ', () => {
    expect(screen).toMatchSnapshot();
  });

  it('input defined ', () => {
    const usernameInput = screen.getByTestId('usernameInput');
    const passwordInput = screen.getByTestId('passwordInput');
    const loginButton = screen.getByTestId('loginButton');

    expect(usernameInput).toBeDefined();
    expect(passwordInput).toBeDefined();
    expect(loginButton).toBeDefined();
  });

  it('username input validated correctly', () => {
    const testInput = 'ftkem2003';

    // 아이디 입력
    const usernameInput = screen.getByTestId('usernameInput');
    fireEvent.changeText(usernameInput, testInput);
    expect(usernameInput.props.value).toBe(testInput);

    // 로그인
    const loginButton = screen.getByTestId('loginButton');
    fireEvent.press(loginButton);

    // 에러 메세지 출력
    const errorMessage = screen.queryByText('잘못된 이메일 형식입니다.');
    expect(errorMessage).toBeDefined();

    const testInput2 = '';

    // 아이디 입력
    fireEvent.changeText(usernameInput, testInput2);
    expect(usernameInput.props.value).toBe(testInput2);

    fireEvent.press(loginButton);

    // 에러 메세지 출력
    const errorMessage2 = screen.queryByText('이메일을 입력해주세요.');
    expect(errorMessage2).toBeDefined();
  });

  it('password input validated correctly', () => {
    const testInput = 'four';

    // 아이디 입력
    const passwordInput = screen.getByTestId('passwordInput');
    fireEvent.changeText(passwordInput, testInput);
    expect(passwordInput.props.value).toBe(testInput);

    // 로그인
    const loginButton = screen.getByTestId('loginButton');
    fireEvent.press(loginButton);

    // 에러 메세지 출력
    const errorMessage = screen.queryByText(
      '비밀번호는 5자리 이상 입력해주세요.',
    );
    expect(errorMessage).toBeDefined();

    const testInput2 = '';

    // 아이디 입력
    fireEvent.changeText(passwordInput, testInput2);
    expect(passwordInput.props.value).toBe(testInput2);

    fireEvent.press(loginButton);

    // 에러 메세지 출력
    const errorMessage2 = screen.queryByText('비밀번호를 입력해주세요.');
    expect(errorMessage2).toBeDefined();
  });
});
