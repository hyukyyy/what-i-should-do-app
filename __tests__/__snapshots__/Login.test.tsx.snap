// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Screen renders correctly  1`] = `
<View
  style={
    {
      "marginVertical": 12,
      "width": "100%",
    }
  }
>
  <View
    style={{}}
  >
    <View
      pointerEvents="none"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 6,
          },
          {
            "top": 0,
          },
          {
            "backgroundColor": "rgba(255, 251, 254, 1)",
            "borderColor": "rgba(121, 116, 126, 1)",
            "borderRadius": 4,
            "borderWidth": 1,
          },
          undefined,
        ]
      }
      testID="text-input-outline"
    />
    <View
      style={
        [
          {
            "paddingBottom": 0,
          },
          {
            "minHeight": 56,
            "paddingTop": 0,
          },
        ]
      }
    >
      <TextInput
        cursorColor="rgba(103, 80, 164, 1)"
        editable={true}
        maxFontSizeMultiplier={1.5}
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        onLayout={[Function]}
        placeholderTextColor="rgba(73, 69, 79, 1)"
        selectionColor="#ef5bae"
        style={
          [
            {
              "margin": 0,
            },
            {
              "height": 56,
            },
            {
              "paddingBottom": 0,
              "paddingTop": 0,
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "letterSpacing": 0.15,
              "lineHeight": undefined,
              "minWidth": 65,
              "paddingHorizontal": 16,
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            false,
            [
              {},
            ],
            undefined,
          ]
        }
        testID="text-input-outlined"
        underlineColorAndroid="transparent"
      >
        <Image
          resizeMode="cover"
          source={
            {
              "testUri": "../../../assets/logo.png",
            }
          }
          style={
            {
              "height": 120,
              "marginBottom": 8,
              "objectFit": "contain",
              "width": 300,
            }
          }
        />
        <Text
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontWeight": "400",
                "letterSpacing": 0,
              },
              {
                "writingDirection": "ltr",
              },
              {
                "color": "#ef5bae",
                "fontSize": 21,
                "fontWeight": "bold",
                "paddingVertical": 12,
              },
            ]
          }
        >
          Welcome back.
        </Text>
        <View
          style={
            {
              "marginVertical": 12,
              "width": "100%",
            }
          }
        >
          <View
            style={{}}
          >
            <View
              pointerEvents="none"
              style={
                [
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 6,
                  },
                  false,
                  {
                    "backgroundColor": "rgba(255, 251, 254, 1)",
                    "borderColor": "rgba(121, 116, 126, 1)",
                    "borderRadius": 4,
                    "borderWidth": 1,
                  },
                  undefined,
                ]
              }
              testID="text-input-outline"
            />
            <View
              style={
                [
                  {
                    "paddingBottom": 0,
                  },
                  {
                    "minHeight": 56,
                    "paddingTop": 8,
                  },
                ]
              }
            >
              <View
                collapsable={false}
                pointerEvents="none"
                style={
                  {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "transform": [
                      {
                        "translateX": 0,
                      },
                    ],
                    "width": 750,
                    "zIndex": 3,
                  }
                }
              >
                <Text
                  collapsable={false}
                  maxFontSizeMultiplier={1.5}
                  numberOfLines={1}
                  style={
                    {
                      "backgroundColor": "rgba(255, 251, 254, 1)",
                      "color": "transparent",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 8,
                      "letterSpacing": 0.15,
                      "lineHeight": undefined,
                      "opacity": -0.6666666666666667,
                      "paddingHorizontal": 0,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 33,
                      "transform": [
                        {
                          "translateX": 0,
                        },
                        {
                          "translateY": 0,
                        },
                        {
                          "scale": 1,
                        },
                        {
                          "scaleY": 1,
                        },
                      ],
                      "width": -16,
                      "writingDirection": "ltr",
                    }
                  }
                  testID="usernameInput-label-background"
                >
                  Email
                </Text>
                <Text
                  collapsable={false}
                  maxFontSizeMultiplier={1.5}
                  numberOfLines={1}
                  onLayout={[Function]}
                  onTextLayout={[Function]}
                  style={
                    {
                      "color": "rgba(103, 80, 164, 1)",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 0,
                      "letterSpacing": 0.15,
                      "lineHeight": undefined,
                      "maxWidth": 73,
                      "opacity": 0,
                      "paddingHorizontal": 16,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 32,
                      "transform": [
                        {
                          "translateX": 0,
                        },
                        {
                          "translateY": 0,
                        },
                        {
                          "scale": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                  testID="usernameInput-label-active"
                >
                  Email
                </Text>
                <Text
                  collapsable={false}
                  maxFontSizeMultiplier={1.5}
                  numberOfLines={1}
                  style={
                    {
                      "color": "rgba(73, 69, 79, 1)",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 0,
                      "letterSpacing": 0.15,
                      "lineHeight": undefined,
                      "maxWidth": 73,
                      "opacity": 0,
                      "paddingHorizontal": 16,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 32,
                      "transform": [
                        {
                          "translateX": 0,
                        },
                        {
                          "translateY": 0,
                        },
                        {
                          "scale": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                  testID="usernameInput-label-inactive"
                >
                  Email
                </Text>
              </View>
              <TextInput
                autoCapitalize="none"
                autoCompleteType="email"
                cursorColor="rgba(103, 80, 164, 1)"
                editable={true}
                keyboardType="email-address"
                maxFontSizeMultiplier={1.5}
                multiline={false}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                onLayout={[Function]}
                placeholder=" "
                placeholderTextColor="rgba(73, 69, 79, 1)"
                returnKeyType="next"
                selectionColor="#ef5bae"
                style={
                  [
                    {
                      "margin": 0,
                    },
                    {
                      "height": 48,
                    },
                    {
                      "paddingBottom": 0,
                      "paddingTop": 0,
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "letterSpacing": 0.15,
                      "lineHeight": undefined,
                      "minWidth": 65,
                      "paddingHorizontal": 16,
                      "textAlign": "left",
                      "textAlignVertical": "center",
                    },
                    false,
                    [
                      {},
                    ],
                    undefined,
                  ]
                }
                testID="usernameInput"
                textContentType="emailAddress"
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
          </View>
        </View>
        <View
          style={
            {
              "marginVertical": 12,
              "width": "100%",
            }
          }
        >
          <View
            style={{}}
          >
            <View
              pointerEvents="none"
              style={
                [
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 6,
                  },
                  false,
                  {
                    "backgroundColor": "rgba(255, 251, 254, 1)",
                    "borderColor": "rgba(121, 116, 126, 1)",
                    "borderRadius": 4,
                    "borderWidth": 1,
                  },
                  undefined,
                ]
              }
              testID="text-input-outline"
            />
            <View
              style={
                [
                  {
                    "paddingBottom": 0,
                  },
                  {
                    "minHeight": 56,
                    "paddingTop": 8,
                  },
                ]
              }
            >
              <View
                collapsable={false}
                pointerEvents="none"
                style={
                  {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "transform": [
                      {
                        "translateX": 0,
                      },
                    ],
                    "width": 750,
                    "zIndex": 3,
                  }
                }
              >
                <Text
                  collapsable={false}
                  maxFontSizeMultiplier={1.5}
                  numberOfLines={1}
                  style={
                    {
                      "backgroundColor": "rgba(255, 251, 254, 1)",
                      "color": "transparent",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 8,
                      "letterSpacing": 0.15,
                      "lineHeight": undefined,
                      "opacity": -0.6666666666666667,
                      "paddingHorizontal": 0,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 33,
                      "transform": [
                        {
                          "translateX": 0,
                        },
                        {
                          "translateY": 0,
                        },
                        {
                          "scale": 1,
                        },
                        {
                          "scaleY": 1,
                        },
                      ],
                      "width": -16,
                      "writingDirection": "ltr",
                    }
                  }
                  testID="passwordInput-label-background"
                >
                  Password
                </Text>
                <Text
                  collapsable={false}
                  maxFontSizeMultiplier={1.5}
                  numberOfLines={1}
                  onLayout={[Function]}
                  onTextLayout={[Function]}
                  style={
                    {
                      "color": "rgba(103, 80, 164, 1)",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 0,
                      "letterSpacing": 0.15,
                      "lineHeight": undefined,
                      "maxWidth": 73,
                      "opacity": 0,
                      "paddingHorizontal": 16,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 32,
                      "transform": [
                        {
                          "translateX": 0,
                        },
                        {
                          "translateY": 0,
                        },
                        {
                          "scale": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                  testID="passwordInput-label-active"
                >
                  Password
                </Text>
                <Text
                  collapsable={false}
                  maxFontSizeMultiplier={1.5}
                  numberOfLines={1}
                  style={
                    {
                      "color": "rgba(73, 69, 79, 1)",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 0,
                      "letterSpacing": 0.15,
                      "lineHeight": undefined,
                      "maxWidth": 73,
                      "opacity": 0,
                      "paddingHorizontal": 16,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 32,
                      "transform": [
                        {
                          "translateX": 0,
                        },
                        {
                          "translateY": 0,
                        },
                        {
                          "scale": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                  testID="passwordInput-label-inactive"
                >
                  Password
                </Text>
              </View>
              <TextInput
                cursorColor="rgba(103, 80, 164, 1)"
                editable={true}
                maxFontSizeMultiplier={1.5}
                multiline={false}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                onLayout={[Function]}
                placeholder=" "
                placeholderTextColor="rgba(73, 69, 79, 1)"
                returnKeyType="done"
                secureTextEntry={true}
                selectionColor="#ef5bae"
                style={
                  [
                    {
                      "margin": 0,
                    },
                    {
                      "height": 48,
                    },
                    {
                      "paddingBottom": 0,
                      "paddingTop": 0,
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "letterSpacing": 0.15,
                      "lineHeight": undefined,
                      "minWidth": 65,
                      "paddingHorizontal": 16,
                      "textAlign": "left",
                      "textAlignVertical": "center",
                    },
                    false,
                    [
                      {},
                    ],
                    undefined,
                  ]
                }
                testID="passwordInput"
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
          </View>
        </View>
        <View
          style={
            {
              "alignItems": "flex-end",
              "marginBottom": 24,
              "width": "100%",
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  {
                    "writingDirection": "ltr",
                  },
                  {
                    "color": "#26bfb5",
                    "fontSize": 13,
                  },
                ]
              }
            >
              Forgot your password?
            </Text>
          </View>
        </View>
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "#ef5bae",
              "borderRadius": 20,
              "marginVertical": 10,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
              "width": "100%",
            }
          }
          testID="loginButton-container-outer-layer"
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "#ef5bae",
                "borderColor": "transparent",
                "borderRadius": 20,
                "borderStyle": "solid",
                "borderWidth": 0,
                "flex": undefined,
                "minWidth": 64,
                "paddingVertical": 2,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
            testID="loginButton-container"
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": false,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "overflow": "hidden",
                  },
                  {
                    "borderRadius": 20,
                  },
                ]
              }
              testID="loginButton"
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    [
                      {
                        "textAlign": "left",
                      },
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "writingDirection": "ltr",
                      },
                      [
                        {
                          "fontFamily": "System",
                          "fontSize": 14,
                          "fontWeight": "500",
                          "letterSpacing": 0.1,
                          "lineHeight": 20,
                        },
                        [
                          {
                            "marginHorizontal": 16,
                            "marginVertical": 9,
                            "textAlign": "center",
                          },
                          false,
                          {
                            "marginHorizontal": 24,
                            "marginVertical": 10,
                          },
                          undefined,
                          false,
                          {
                            "color": "rgba(255, 255, 255, 1)",
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          {
                            "fontSize": 15,
                            "fontWeight": "bold",
                            "lineHeight": 26,
                          },
                        ],
                      ],
                    ]
                  }
                  testID="loginButton-text"
                >
                  Login
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            {
              "flexDirection": "row",
              "marginTop": 4,
            }
          }
        >
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                },
                {
                  "writingDirection": "ltr",
                },
                undefined,
              ]
            }
          >
            Don't have an account? 
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  {
                    "writingDirection": "ltr",
                  },
                  {
                    "color": "#ef5bae",
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              Sign up
            </Text>
          </View>
        </View>
      </TextInput>
    </View>
  </View>
</View>
`;
